src/Entities/Entity.cpp

```diff
@@ -2285,3 +2285,98 @@ void cEntity::BroadcastLeashedMobs()
 		}
 	}
 }
+
+
+
+
+
+void cEntity::BroadcastDeathMessage(TakeDamageInfo & a_TDI)
+{
+	cPluginManager * PluginManager = cRoot::Get()->GetPluginManager();
+
+	AString Name = "";
+	if (IsPlayer())
+	{
+		cPlayer * Player = static_cast<cPlayer *>(this);
+		Name = Player->GetName();
+	}
+	else
+	{
+		cMonster * Monster = static_cast<cMonster *>(this);
```

Comment:
We shouldn't be doing an unsafe cast here - what if somebody calls this with a non-mob entity?

Reply:
This is changed by adding a check to the IsMob method and qutting if the entity is neither a mob nor a player.